packages/material-ui/src/Select/SelectInput.js

```diff
@@ -16,6 +16,7 @@ import { isFilled } from '../InputBase/utils';
 import styled, { slotShouldForwardProp } from '../styles/styled';
 import useForkRef from '../utils/useForkRef';
 import useControlled from '../utils/useControlled';
+import areEqualValues from '../utils/areEqualValues';
```

Comment:
I would vote against this abstraction. It seems too specific and too early to introduce. 
```suggestion
```